File: src/main/java/org/jukeboxmc/block/BlockSoulFire.java
package org.jukeboxmc.block;

import org.jukeboxmc.item.ItemSoulFire;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public class BlockSoulFire extends Block {

    public BlockSoulFire() {
        super( "minecraft:soul_fire" );
    }

    @Override
    public ItemSoulFire toItem() {
        return new ItemSoulFire();
    }

    @Override
    public BlockType getType() {
        return BlockType.SOUL_FIRE;
    }

    @Override
    public boolean canPassThrough() {
        return true;
    }

    public void setAge( int value ) {
        this.setState( "age", value );
    }

    public int getAge() {
        return this.stateExists( "age" ) ? this.getIntState( "age" ) : 0;
    }
}

File: JukeboxMC-Server/src/main/kotlin/org/jukeboxmc/server/block/behavior/BlockSoulFire.kt
package org.jukeboxmc.server.block.behavior

import org.cloudburstmc.nbt.NbtMap
import org.jukeboxmc.api.Identifier
import org.jukeboxmc.api.block.SoulFire
import org.jukeboxmc.server.block.JukeboxBlock

class BlockSoulFire(identifier: Identifier, blockStates: NbtMap?) : JukeboxBlock(identifier, blockStates), SoulFire {

    override fun canPassThrough(): Boolean {
        return true
    }

    override fun getAge(): Int {
        return this.getIntState("age")
    }

    override fun setAge(value: Int): SoulFire {
        return this.setState("age", value)
    }
}